metrics/resource_view_test.go

```diff
@@ -170,12 +173,16 @@ func TestAllMetersExpiration(t *testing.T) {
 
 	// Expire the second entry
 	fakeClock.Step(9 * time.Minute) // t+12m
-	time.Sleep(time.Second)         // Wait a second on the wallclock, so that the cleanup thread has time to finish a loop
-	allMeters.lock.Lock()
+	_ = wait.PollImmediate(100*time.Millisecond, 5*time.Second, func() (bool, error) {
+		// Non-expiring defaultMeter should be available
+		allMeters.lock.Lock()
+		defer allMeters.lock.Unlock()
+		return len(allMeters.meters) == 2, nil
+	})
+
 	if len(allMeters.meters) != 2 {
```

Comment:
Wonder why we need a lock for the other accesses to `len(allMeters.meters)` (e.g. L179 and L157) but we dont here 🤔 (race detector isn't complaining, though, so maybe it's fine 🤷)

Reply:
Because of the nature of the test there should not be a race we progress the clock manually and when poll ends we should read the right value. No other goroutine is expected to write anything in allMeters at that point. For the sake of readability and having less confusion I can change it and do a lock/unlock there.